MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm

```diff
@@ -1275,12 +1275,16 @@ static VkFormat mvkFormatFromOutput(const SPIRVShaderOutput& output) {
     if (pixFmts->isDepthFormat(mtlDSFormat)) { plDesc.depthAttachmentPixelFormat = mtlDSFormat; }
     if (pixFmts->isStencilFormat(mtlDSFormat)) { plDesc.stencilAttachmentPixelFormat = mtlDSFormat; }
 
-    // In Vulkan, it's perfectly valid to render with no attachments. Not so in Metal.
-	// If we have no attachments, then we'll have to add a dummy attachment.
-    if (!caCnt && !pixFmts->isDepthFormat(mtlDSFormat) && !pixFmts->isStencilFormat(mtlDSFormat)) {
-        MTLRenderPipelineColorAttachmentDescriptor* colorDesc = plDesc.colorAttachments[0];
-        colorDesc.pixelFormat = MTLPixelFormatR8Unorm;
-        colorDesc.writeMask = MTLColorWriteMaskNone;
+    // In Vulkan, it's perfectly valid to render with no attachments. In Metal we need to detect
+    // support for it. If we have no attachments, then we may have to add a dummy attachment.
+    @autoreleasepool {
+        if (!caCnt && !pixFmts->isDepthFormat(mtlDSFormat) && !pixFmts->isStencilFormat(mtlDSFormat) &&
+            ![[MTLRenderPassDescriptor renderPassDescriptor] respondsToSelector: @selector(defaultRasterSampleCount)]) {
```

Comment:
This would be more consistently handled by creating an entry at the **_end_** of `MVKPhysicalDeviceMetalFeatures` (maybe `renderWithoutAttachments`?), and set its it to `true` in `MVKPhysicalDevice::initMetalFeatures()`, in the `#if MVK_IOS` and `#if MVK_MACOS` sections, using `mvkOSVersionIsAtLeast(11.0)` and `mvkOSVersionIsAtLeast(10.15)` (which already exists there), , respectively...and then testing for that flag here.

Reply:
Agreed, that's better.